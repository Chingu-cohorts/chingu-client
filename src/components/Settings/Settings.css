.settings {
  margin-top: 3rem;
  min-height: 65vh;
}
