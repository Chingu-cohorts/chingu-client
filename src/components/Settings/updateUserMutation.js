import gql from "graphql-tag";

const updateUser = gql`
  mutation updateUser(
    $first_name: String!
    $last_name: String!
    $bio: String
    $github_url: String!
    $twitter_url: String
    $linkedin_url: String
    $portfolio_url: String
    $website_url: String
    $blog_url: String
  ) {
    updateUser(
      user_data: {
        first_name: $first_name
        last_name: $last_name
        bio: $bio
        github_url: $github_url
        twitter_url: $twitter_url
        linkedin_url: $linkedin_url
        portfolio_url: $portfolio_url
        website_url: $website_url
        blog_url: $blog_url
      }
    ) {
      first_name
    }
  }
`;

export default updateUser;
