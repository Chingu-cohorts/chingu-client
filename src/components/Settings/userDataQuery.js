import gql from "graphql-tag";

const userDataQuery = gql`
  query userData {
    user {
      id
      first_name
      last_name
      bio
      github_url
      twitter_url
      linkedin_url
      portfolio_url
      website_url
      blog_url
    }
  }
`;

export default userDataQuery;
