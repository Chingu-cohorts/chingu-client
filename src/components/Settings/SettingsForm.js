import React, { Component } from "react";
import { graphql } from "react-apollo";
import { Form, Tabs, Input, Button } from "antd";

import updateUserMutation from "./updateUserMutation";

const { TabPane } = Tabs;
const FormItem = Form.Item;

class SettingsForm extends Component {
  componentDidMount() {
    let {
      first_name,
      last_name,
      bio,
      github_url,
      twitter_url,
      linkedin_url,
      portfolio_url,
      website_url,
      blog_url
    } = this.props;

    this.props.form.setFieldsValue({
      first_name,
      last_name,
      bio,
      github_url,
      twitter_url,
      linkedin_url,
      portfolio_url,
      website_url,
      blog_url
    });
  }

  handleProfileSubmit = e => {
    e.preventDefault();

    this.props.form.validateFields((err, values) => {
      if (!err) {
        let {
          first_name,
          last_name,
          bio,
          github_url,
          twitter_url,
          linkedin_url,
          portfolio_url,
          website_url,
          blog_url
        } = values;

        console.log(values);

        this.props
          .mutate({
            variables: {
              first_name: first_name,
              last_name: last_name,
              bio: bio,
              github_url: github_url,
              twitter_url: twitter_url,
              linkedin_url: linkedin_url,
              portfolio_url: portfolio_url,
              website_url: website_url,
              blog_url: blog_url
            }
          })
          .then(({ data }) => {
            console.log(data);
          })
          .catch(err => {
            console.error(err);
          });
      }
    });
  };

  render() {
    const { getFieldDecorator } = this.props.form;
    return (
      <Tabs defaultActiveKey="1" tabPosition="left">
        <TabPane tab="Profile" key="1">
          <Form onSubmit={this.handleProfileSubmit}>
            <FormItem
              label="First name"
              labelCol={{ span: 4 }}
              wrapperCol={{ span: 10 }}
            >
              {getFieldDecorator("first_name", {
                rules: [
                  {
                    required: true,
                    message: "Please enter your first name"
                  }
                ]
              })(<Input placeholder="First name" />)}
            </FormItem>
            <FormItem
              label="Last name"
              labelCol={{ span: 4 }}
              wrapperCol={{ span: 10 }}
            >
              {getFieldDecorator("last_name", {
                rules: [
                  {
                    required: true,
                    message: "Please enter your last name"
                  }
                ]
              })(<Input placeholder="Last name" />)}
            </FormItem>
            <FormItem
              label="Bio"
              labelCol={{ span: 4 }}
              wrapperCol={{ span: 10 }}
            >
              {getFieldDecorator("bio")(<Input.TextArea rows={4} />)}
            </FormItem>
            <FormItem>
              <Button type="primary" htmlType="submit">
                Update profile
              </Button>
            </FormItem>
          </Form>
        </TabPane>
        <TabPane tab="Networks" key="2">
          <Form onSubmit={this.handleProfileSubmit}>
            <FormItem
              label="GitHub"
              labelCol={{ span: 4 }}
              wrapperCol={{ span: 10 }}
            >
              {getFieldDecorator("github_url", {
                rules: [
                  {
                    required: true,
                    message: "Please enter your GitHub URL"
                  }
                ]
              })(<Input placeholder="https://github.com/chingu-x" />)}
            </FormItem>
            <FormItem
              label="Twitter"
              labelCol={{ span: 4 }}
              wrapperCol={{ span: 10 }}
            >
              {getFieldDecorator("twitter_url")(
                <Input placeholder="https://twitter.com/Chingu-collabs" />
              )}
            </FormItem>
            <FormItem
              label="LinkedIn"
              labelCol={{ span: 4 }}
              wrapperCol={{ span: 10 }}
            >
              {getFieldDecorator("linkedin_url")(
                <Input placeholder="https://linkedin.com/chingu" />
              )}
            </FormItem>
            <FormItem
              label="Portfolio"
              labelCol={{ span: 4 }}
              wrapperCol={{ span: 10 }}
            >
              {getFieldDecorator("portfolio_url")(
                <Input placeholder="https://portfolio.com" />
              )}
            </FormItem>
            <FormItem
              label="Website"
              labelCol={{ span: 4 }}
              wrapperCol={{ span: 10 }}
            >
              {getFieldDecorator("website_url")(
                <Input placeholder="https://chingu.io" />
              )}
            </FormItem>
            <FormItem
              label="Blog"
              labelCol={{ span: 4 }}
              wrapperCol={{ span: 10 }}
            >
              {getFieldDecorator("blog_url")(
                <Input placeholder="https://medium.com/chingu" />
              )}
            </FormItem>
            <FormItem>
              <Button type="primary" htmlType="submit">
                Update profile
              </Button>
            </FormItem>
          </Form>
        </TabPane>
      </Tabs>
    );
  }
}

export const WrappedForm = Form.create()(SettingsForm);

export default graphql(updateUserMutation)(WrappedForm);
