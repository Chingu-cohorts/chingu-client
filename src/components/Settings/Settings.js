import React, { Component } from "react";
import { graphql } from "react-apollo";
import { Layout, Row, Col, Spin } from "antd";

import userDataQuery from "./userDataQuery";
import SettingsForm from "./SettingsForm";
import "./Settings.css";

const { Content } = Layout;

export class Settings extends Component {
  render() {
    let { loading, error, user } = this.props.data;
    if (loading) {
      return (
        <Content>
          <div className="text-center">
            <Spin />
          </div>
        </Content>
      );
    } else if (error) {
      return (
        <Content>
          <h1>An error ocurred</h1>
        </Content>
      );
    }
    return (
      <Content>
        <div className="settings">
          <h1 className="text-center">Update your settings and profile</h1>
          <Row type="flex" justify="center">
            <Col sm={22} md={18} lg={16}>
              <SettingsForm {...user} />
            </Col>
          </Row>
        </div>
      </Content>
    );
  }
}

export default graphql(userDataQuery)(Settings);
