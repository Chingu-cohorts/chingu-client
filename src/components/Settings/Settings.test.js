import React from "react";
import { shallow } from "enzyme";

import { Settings } from "./Settings";

describe("Settings Component", () => {
  const props = {
    data: {
      loading: false,
      error: false,
      user: {
        first_name: "Andrés",
        last_name: "Pérez"
      }
    }
  };
  it("renders without crashing", () => {
    const wrapper = shallow(<Settings {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
