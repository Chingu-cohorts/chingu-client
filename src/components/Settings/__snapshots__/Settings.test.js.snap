// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Component renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Settings
    data={
        Object {
            "error": false,
            "loading": false,
            "user": Object {
              "first_name": "Andrés",
              "last_name": "Pérez",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="settings"
>
        <h1
                className="text-center"
        >
                Update your settings and profile
        </h1>
        <Row
                gutter={0}
                justify="center"
                style={
                        Object {
                                "marginTop": "3rem",
                              }
                }
                type="flex"
        >
                <Col
                        lg={14}
                        md={14}
                        offset={4}
                        sm={16}
                >
                        <Apollo(Form(WrappedComponent))
                                first_name="Andrés"
                                last_name="Pérez"
                        />
                </Col>
        </Row>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="text-center"
>
            Update your settings and profile
</h1>,
          <Row
            gutter={0}
            justify="center"
            style={
                        Object {
                                    "marginTop": "3rem",
                                  }
            }
            type="flex"
>
            <Col
                        lg={14}
                        md={14}
                        offset={4}
                        sm={16}
            >
                        <Apollo(Form(WrappedComponent))
                                    first_name="Andrés"
                                    last_name="Pérez"
                        />
            </Col>
</Row>,
        ],
        "className": "settings",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Update your settings and profile",
            "className": "text-center",
          },
          "ref": null,
          "rendered": "Update your settings and profile",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Col
              lg={14}
              md={14}
              offset={4}
              sm={16}
>
              <Apollo(Form(WrappedComponent))
                            first_name="Andrés"
                            last_name="Pérez"
              />
</Col>,
            "gutter": 0,
            "justify": "center",
            "style": Object {
              "marginTop": "3rem",
            },
            "type": "flex",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Apollo(Form(WrappedComponent))
                first_name="Andrés"
                last_name="Pérez"
/>,
              "lg": 14,
              "md": 14,
              "offset": 4,
              "sm": 16,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "first_name": "Andrés",
                "last_name": "Pérez",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="settings"
>
          <h1
                    className="text-center"
          >
                    Update your settings and profile
          </h1>
          <Row
                    gutter={0}
                    justify="center"
                    style={
                              Object {
                                        "marginTop": "3rem",
                                      }
                    }
                    type="flex"
          >
                    <Col
                              lg={14}
                              md={14}
                              offset={4}
                              sm={16}
                    >
                              <Apollo(Form(WrappedComponent))
                                        first_name="Andrés"
                                        last_name="Pérez"
                              />
                    </Col>
          </Row>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="text-center"
>
              Update your settings and profile
</h1>,
            <Row
              gutter={0}
              justify="center"
              style={
                            Object {
                                          "marginTop": "3rem",
                                        }
              }
              type="flex"
>
              <Col
                            lg={14}
                            md={14}
                            offset={4}
                            sm={16}
              >
                            <Apollo(Form(WrappedComponent))
                                          first_name="Andrés"
                                          last_name="Pérez"
                            />
              </Col>
</Row>,
          ],
          "className": "settings",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Update your settings and profile",
              "className": "text-center",
            },
            "ref": null,
            "rendered": "Update your settings and profile",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Col
                lg={14}
                md={14}
                offset={4}
                sm={16}
>
                <Apollo(Form(WrappedComponent))
                                first_name="Andrés"
                                last_name="Pérez"
                />
</Col>,
              "gutter": 0,
              "justify": "center",
              "style": Object {
                "marginTop": "3rem",
              },
              "type": "flex",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Apollo(Form(WrappedComponent))
                  first_name="Andrés"
                  last_name="Pérez"
/>,
                "lg": 14,
                "md": 14,
                "offset": 4,
                "sm": 16,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "first_name": "Andrés",
                  "last_name": "Pérez",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
